codec-http/src/main/java/io/netty/handler/codec/http/HttpVersion.java

```diff
@@ -138,21 +133,32 @@ public HttpVersion(String text, boolean keepAliveDefault) {
             majorVersion = toDecimal(text.charAt(5));
             minorVersion = toDecimal(text.charAt(7));
         } else {
-            Matcher m = VERSION_PATTERN.matcher(text);
-            if (!m.matches()) {
+            int slashIndex = text.indexOf('/');
+            int dotIndex = text.indexOf('.', slashIndex + 1);
+
+            if (slashIndex <= 0 || dotIndex <= slashIndex + 1 || dotIndex >= text.length() - 1) {
                 throw new IllegalArgumentException("invalid version format: " + text);
             }
 
-            protocolName = m.group(1);
-            majorVersion = Integer.parseInt(m.group(2));
-            minorVersion = Integer.parseInt(m.group(3));
+            protocolName = text.substring(0, slashIndex);
```

Comment:
To preserve the existing behavior I think we should also check that the `protocolName` only contain non-whitespace characters.

Reply:
@doom369 please address and maybe also add a test ?